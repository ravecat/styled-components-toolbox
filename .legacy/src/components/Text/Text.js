import styled from "styled-components";

import { theme } from "common/theme";
import BaseComponent from "components/BaseComponent";

const Text = styled(BaseComponent).attrs({
  as: "span"
})`
  color: ${theme.primaryTextColor};
  font-size: ${({ size }) => size}px;
  font-style: ${({ italic }) => italic && `italic`};
  text-transform: ${({ uppercase }) => uppercase && `uppercase`};
  line-height: ${({ lheight }) => lheight || theme.lineHeight};
  font-weight: ${({ bold }) => bold && `700`};
  text-decoration: ${({ underline }) => underline && `underline`};
`;

export default Text;
