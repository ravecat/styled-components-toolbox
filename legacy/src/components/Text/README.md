## Text

A simple text component.

### API

| Property    | Description               | Type    | Default |
| ----------- | ------------------------- | ------- | ------- |
| _bold_      | set the bold font style   | boolean | `false` |
| _italic_    | set the italic style      | boolean | `false` |
| _underline_ | set the underline style   | boolean | `false` |
| _uppercase_ | convert text to uppercase | boolean | `false` |
