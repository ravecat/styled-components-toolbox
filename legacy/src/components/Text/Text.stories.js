import * as React from "react";
import { text, boolean, select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Text from "./index";

storiesOf("Text", module).add("Default", () => (
  <Text
    bold={boolean("bold", false)}
    italic={boolean("italic", false)}
    lheight={select("lheight", [0.5, 1, 1.5, 2], 1)}
    size={select("size", [6, 10, 14, 18, 22, 26], 14)}
    underline={boolean("underline", false)}
    uppercase={boolean("uppecase", false)}
  >
    {text("children", "Text")}
  </Text>
));
