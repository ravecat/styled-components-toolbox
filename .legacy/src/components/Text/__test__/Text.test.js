import React from "react";
import renderer from "react-test-renderer";

import Text from "../index";

it("<Text> renders correctly", () => {
  const tree = renderer
    .create(
      <Text bold italic underline uppercase>
        Text
      </Text>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
