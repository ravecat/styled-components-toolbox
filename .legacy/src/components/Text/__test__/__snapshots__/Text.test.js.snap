// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders correctly 1`] = `
<span
  className="sc-bdVaJa sc-bwzfXH ijzeAa"
>
  Text
</span>
`;
